packages/viewer/src/lib/walk-mode/controller.ts

```diff
@@ -0,0 +1,161 @@
+import { Disposable } from '@vertexvis/utils';
+
+import { KeyBinding } from '../keyBinding';
+import {
+  ViewerTeleportMode,
+  ViewerWalkModeConfiguration,
+  ViewerWalkModeOperation,
+  WalkModeModel,
+} from './model';
+
+export class WalkModeController {
+  private excludeTagNames: Array<string> = [
+    'VERTEX-SCENE-TREE',
+    'VERTEX-SCENE-TREE-SEARCH',
+    'VERTEX-VIEWER-PIN-TOOL',
+    'INPUT',
+    'TEXTAREA',
+  ];
+
+  private excludePredicates: Array<(el: Element) => boolean> = [];
+
+  public constructor(private model: WalkModeModel) {
+    this.updateModelExclusions();
+  }
+
+  /**
+   * Sets whether downstream walk mode interaction handlers are enabled.
+   * Setting this value to `false` will remove all event listeners for
+   * the interactions, and setting this value to `true` will add or
+   * re-add the event listeners.
+   */
+  public setEnabled(enabled: boolean): void {
+    this.model.setEnabled(enabled);
+  }
+
+  /**
+   * Sets the `ViewerTeleportMode` to be used with a `<vertex-viewer-teleport-tool>`.
+   */
+  public setTeleportMode(mode?: ViewerTeleportMode): void {
+    this.model.setTeleportMode(mode);
+  }
+
+  /**
+   * Updates the configuration for downstream walk mode interaction handlers.
+   *
+   * `teleportHeightScalar` - scalar used for fine-tuning the distance to offset
+   * the camera from a surface when performing a `teleport-and-align`. A larger
+   * number here will result in a shorter distance from the surface and vice-versa.
+   * Defaults to `8.5`.
```

Comment:
It's accurate with the current math, but I'm not sure that it's intuitive that a larger _teleportHeightScalar_ will place the user closer to the ground.

Reply:
Yeah, that's fair. My thought with changing the way this value worked was primarily to have a whole number instead of a decimal, but I'll rethink this a bit to see if we can get the opposite behavior where a higher number results in a larger distance from the ground

Reply:
That's fair, I agree that having a whole number instead of a decimal would be nice. If it's too much work, we can leave it as it is.